src/services/searchPatient.ts

```diff
@@ -36,9 +39,19 @@ export const searchPatient = async (
     let searchByIdentifier = ''
     let filterByService = ''
     if (groupId) {
-      const services = (await getServices(groupId)).filter((service) => undefined !== service.id)
-      const serviceIds = services.map((service) => service.id)
-      filterByService = `&_has:Encounter:subject:service-provider=${serviceIds.join(',')}`
+      const cohortResponse = await apiBackCohort.get<Back_API_Response<Cohort>>(
+        `/explorations/cohorts/?fhir_group_id=${groupId}`
+      )
+      const requestId = head(cohortResponse.data.results)?.request_id
+      const snapshotResponse = await apiBackCohort.get<Back_API_Response<Snapshot>>(
+        `/explorations/request-query-snapshots/?request_id=${requestId}`
+      )
+      const query_json = head(snapshotResponse.data.results)?.serialized_query
+      if (query_json) {
+        const query: Query = JSON.parse(query_json)
+        const perimeters = query.sourcePopulation.caresiteCohortList
+        filterByService = `&_has:Encounter:subject:service-provider=${perimeters.join(',')}`
```

Comment:
we could add theperimeter in the group "characteristic" attribute to avoid fetching them from the back (cf our discussion)